### DISCUSSION TITLE
How can i get logs of requests hasura sends to action endpoints?

### DISCUSSION THREAD

**Discussion by wemersonrv** [hasura/graphql-engine#7212] [Question] [answered]
Hello, 

How can i get logs of request hasura sends to action endoints?

I need to know the exact content of requests hasura is sending to an action endpoint.

**Comment by wemersonrv** [comment]
Just to complement. The logs i get via docker, i need to know the exact data is sent on the request.

My problem is that the action handler is a Laravel endpoint and when i sent the request direct to the endpoint (without hasura, it works fine)... but when i use the action, something in the request object is not recognize by laravel. So i need to get an exact copy of the entire request data to compare.

**Comment by praveenweb** [comment, marked as answer]
@wemersonrv - If you would like to know the exact payload format that is being sent in the request to an Action, check below:

```
{
  "action": {
    "name": "<action-name>"
  },
  "input": {
    "arg1": "<value>",
    "arg2": "<value>"
  },
  "session_variables": {
    "x-hasura-user-id": "<session-user-id>",
    "x-hasura-role": "<session-user-role>"
  }
}
```

This is the request body that will be received by your laravel endpoint. You will probably need to transform your existing laravel handler to parse this body.

You can learn more in our [docs](https://hasura.io/docs/latest/graphql/core/actions/action-handlers.html#http-handler)

**Reply by wemersonrv** [reply]
Thanks @praveenweb
